include/zim/archive.h

```diff
@@ -75,6 +75,16 @@ namespace zim
        */
       explicit Archive(int fd);
 
+      /** Archive constructor.
+       *
+       *  Construct an archive from a descriptor of a file with an embedded ZIM
+       *  archive inside.
+       *
+       *  @param fd The descriptor of a seekable file with a continuous segment
+       *  representing a complete ZIM archive.
+       */
+      Archive(int fd, offset_type offset, size_type size);
```

Comment:
I wonder if this is the good api.

Xapian provide a way to open a fd at a specific offset by simply providing `Database(int fd)` constructor. It is to the user of library to seek to the right position before passing the fd.

And even if we provide a `offset` parameter. The fd may still be positioned at a specific offset. We should handle this the proper way (seek it to 0 all the time or taking into account the current position) and document this.

Reply:
I chose to document the current API